### PR TITLE
[Feature] Support Ernie FP8 on sm100

### PR DESCRIPTION
**Pull request by zoooo0820**
## Motivation

> :bulb: If this PR is a Cherry Pick, the PR title needs to follow the format by adding the [Cherry-Pick] label at the very beginning and appending the original PR ID at the end. For example, [Cherry-Pick][CI] Add check trigger and logic(#5191)

> :bulb: 如若此PR是Cherry Pick，PR标题需遵循格式，在最开始加上[Cherry-Pick]标签，以及最后面加上原PR ID，例如[Cherry-Pick][CI] Add check trigger and logic(#5191) 

This PR is going to support Blockwise-FP8 inference on SM100.

## Modifications

1. Because the old version of DeepGEMM does not support sm100, we use the newer version DeepGEMM for sm100. And this dependency will be switched to PaddleFleet in the future. 
2. FP8 gemm / group_gemm on sm100 will use ue8m0 scale on sm100. 
3. For EP mode, FP8 DeepEP low_latency_dispatch will use ue8m0 scale on sm100.
4. For activation quantization, we use `paddle.incubate.nn.functional.fp8_quant_blockwise` instead of `per_token_quant` / `per_token_quant_padding`. Add  support ue8m0 for `masked_per_token_quant`, using in EP Decoding phase.

## Usage or Command

**requires paddlepaddle develop after(include) 20260119**

For DeepGEMM installtion: https://github.com/PFCCLab/DeepGEMM
For DeepEP installtion: https://github.com/PFCCLab/DeepEP

## Checklist

- [x] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [x] Format your code, run `pre-commit` before commit.
- [x] Add unit tests. Please write the reason in this PR if no unit tests.
- [x] Provide accuracy results.
- [ ] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

File: fastdeploy/model_executor/layers/quantization/fp8_utils.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+
+def ceil_div(x: int, y: int) -> int:
+    return (x + y - 1) // y
+
+
+def _get_mn_major_tma_aligned_packed_ue8m0_tensor_torch_impl(
+    x: paddle.Tensor,
+):
+    """将FP32张量转换为TMA对齐的packed UE8M0格式张量"""
```
Comment (RichardWooSJTU) [inline fastdeploy/model_executor/layers/quantization/fp8_utils.py:41]: 是否考虑换成英文注释

File: fastdeploy/model_executor/layers/utils.py [hunk 2]
```diff
@@ -254,7 +254,10 @@ def per_block_cast_to_fp8(x: Tensor, block_size: list = [128, 128]) -> Tuple[Ten
     Only used in deep_gemm block wise quant weight.
     copy from FastDeploy/custom_ops/gpu_ops/fp8_deep_gemm/tests/test_core.py.
     """
-    from fastdeploy.model_executor.ops.gpu.deep_gemm import ceil_div
+    try:
+        from deep_gemm import ceil_div
+    except ModuleNotFoundError:
```
Comment (RichardWooSJTU) [inline fastdeploy/model_executor/layers/utils.py:259]: 最好打印出warning吧

**Review by RichardWooSJTU — APPROVED** [review]

File: fastdeploy/model_executor/layers/utils.py (hunk 2 above)
Comment (zoooo0820) [reply fastdeploy/model_executor/layers/utils.py:259]: 好的，这两个问题在后续PR一并修复
